### PR TITLE
[Feature:InstructorUI] Grade override

### LINKED ISSUE #2982: Grade Override: e.g. makeup exam data entry
most of the class takes an exam with a complex rubric.
a couple students take a different exam as makeup, which the instructor grades outside of submitty.
Provide a way to specify just the total for each student as makeup.

possible database changes?
Perhaps a new table for grade override?
gradeableid, userid, score, string message
possibly also a scanned file?  (graded by hand?)

Then when displaying the grade, if there is an entry in the grade override for that student for that score, display that instead.

And grade summaries needs to use this too.

### PR DESCRIPTION
**Pull request by drumilpatel2000**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?

Fixes #2982 .Presently instructor have to adjust in complex rubric for overriding grades. 

### What is the new behavior?
After this PR a new page will be added where overridden grades can be added.
##### Overridden Grades are set in "Grade Override" Page 
Screenshot from 2019-07-06 02-42-06
##### Yellow color (Cadmium Yellow) banner is used to show message to student that grades are overriden 
Screenshot from 2019-07-06 02-42-25
##### "Overridden" is set for gradeable with overridden grades   
Screenshot from 2019-07-06 02-42-53
Screenshot from 2019-07-06 02-43-12
##### "Overridden Grades" is shown in yellow background when grades are overridden in Ta grading
Screenshot from 2019-07-06 02-44-14
##### Grade Summaries
Screenshot from 2019-07-06 02-48-09
##### Grade Summaries in CSV
Screenshot from 2019-07-06 03-14-04

### CODE REVIEW

**Comment by bmcutler** [comment]
I overrode the grades of 2 users on the "closed homework", all looked good, until I went to generate grade summaries and I got the error page, and this info in
/var/local/submitty/logs/site_errors:




00:01:49 07/10/2019 - FATAL ERROR
Error (Code: 0) thrown in /usr/local/submitty/site/app/controllers/admin/ReportController.php (Line 340) by:
        if ($ldi->getStatus() === LateDayInfo::STATUS_OVERRIDDEN){


Message:
Call to a member function getStatus() on null

Stack Trace:
#0 /usr/local/submitty/site/app/controllers/admin/ReportController.php(309): app\controllers\admin\ReportController->generat
eGradeSummary(app\models\gradeable\GradedGradeable, app\models\User, app\models\gradeable\LateDays)
#1 /usr/local/submitty/site/app/controllers/admin/ReportController.php(76): app\controllers\admin\ReportController->saveUser
ToFile('/var/local/submitty/courses/s19/sample/reports/all_grades', app\models\User, Array, app\models\gradeable\LateDays)
#2 /usr/local/submitty/site/app/controllers/admin/ReportController.php(214): app\controllers\admin\ReportController->app\con
trollers\admin\{closure}(app\models\User, Array, app\models\gradeable\LateDays)
#3 /usr/local/submitty/site/app/controllers/admin/ReportController.php(221): app\controllers\admin\ReportController->app\con
trollers\admin\{closure}(Array, app\models\User, Array, Array, Closure)
#4 /usr/local/submitty/site/app/controllers/admin/ReportController.php(78): app\controllers\admin\ReportController->generate
ReportInternal(Array, Array, Closure)
#5 unknown file(unknown line): app\controllers\admin\ReportController->generateGradeSummaries()
#6 /usr/local/submitty/site/app/libraries/routers/WebRouter.php(143): call_user_func_array(Array, Array)
#7 /usr/local/submitty/site/public/index.php(344): app\libraries\routers\WebRouter->run()

URL: http://192.168.56.111//s19/sample/reports/summaries
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

**Comment by drumilpatel2000** [comment]
@bmcutler I have solved the particular problem. But I came across a new problem delete is not executed properly

File: migration/migrator/migrations/course/20190703055948_grade_override.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+def up(config, database, semester, course):
+    database.execute("CREATE TABLE grade_override (user_id character varying(255) NOT NULL,g_id character varying(255) NOT NULL,marks integer NOT NULL,comment character varying )")
+    
+    database.execute("ALTER TABLE ONLY grade_override ADD CONSTRAINT grade_override_pkey PRIMARY KEY (user_id, g_id);")
+    database.execute("ALTER TABLE ONLY grade_override ADD CONSTRAINT grade_override_g_id_fkey FOREIGN KEY (g_id) REFERENCES gradeable(g_id) ON DELETE CASCADE;")
+    database.execute("ALTER TABLE ONLY grade_override ADD CONSTRAINT grade_override_user_id_fkey FOREIGN KEY (user_id) REFERENCES users(user_id) ON UPDATE CASCADE;")
+    
+
+
+def down(config, database, semester, course):
```
Comment (thef1stofhername) [inline migration/migrator/migrations/course/20190703055948_grade_override.py]: you need a down migration

Comment (drumilpatel2000) [reply migration/migrator/migrations/course/20190703055948_grade_override.py]: Thanks @thef1stofhername . Probably dropping the table is the best options

**Comment by bmcutler** [comment]
2 items:

* I get an error when I try to put a decimal in the score, e.g. "1.5"

* Can you make the student id box work like the id selection box on the extensions page, specifically that you can type the username, first name, or last name and it will search for all matches in any of those fields, and it will also display the first & last name along with the student id.  Right now it does the above but only for the id.

**Comment by drumilpatel2000** [comment]
@bmcutler can review it again

**Comment by bmcutler** [comment]
I still get "error parsing data" when I try to put a decimal in the form.

**Comment by bmcutler** [comment]
I had stale migration from previous version

**Pull request merged**